### PR TITLE
BaSP-TG-109: FE Employee timesheets screen

### PR DESCRIPTION
**Pull request by Johsua30**
Added Month to timesheet table(Will probably change it to period).
Made it so upon creating a new timesheet it saves to the timesheet array inside the employee.
A timesheet can now be deleted by the employee.
The new timesheet starts with no tasks and approved: false

### CODE REVIEW

File: src/Components/Employee/Timesheet/AddTimesheet/index.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  task: Joi.string().allow(''),
+  approved: Joi.boolean().required()
+});
+
+const FormAdd = ({ closeModalForm }) => {
+  const dispatch = useDispatch();
+  const [message, setMessage] = useState('');
+  const [showMessageModal, setShowMessageModal] = useState(false);
+  const projects = useSelector((state) => state.projects.list);
+  // const tasks = useSelector((state) => state.tasks.list);
```
Comment (agusting19) [inline src/Components/Employee/Timesheet/AddTimesheet/index.jsx:26]: Commented line

File: src/Components/Employee/Timesheet/AddTimesheet/form.module.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.form {
+    flex: 1;
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: flex-end;
+    align-items: flex-end;
+    align-content: space-between;
+    row-gap: 10px;
```
Comment (agusting19) [inline src/Components/Employee/Timesheet/AddTimesheet/form.module.css:9]: Maybe we could add some margin botton, the container is attached to the footer

File: src/Components/Employee/Timesheet/AddTimesheet/form.module.css [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.form {
+    flex: 1;
+    display: flex;
+    flex-direction: row;
```
Comment (agusting19) [inline src/Components/Employee/Timesheet/AddTimesheet/form.module.css:4]: line 1 and 3 are unnecessary

**Review by agusting19 — COMMENTED** [review]
The add button could be an text button with "add" label. Also when i try to add a new timesheet tells me that the timesheet already exist (even if it doesn't)

File: src/Components/Employee/Timesheet/index.jsx [hunk 4]
```diff
@@ -11,22 +26,98 @@ function Timesheet() {
...
+        titles={['Project', 'PMs approval', 'Worked hours', 'Period']}
+        delAction={(id) => {
+          setTimeSheetId(id);
+          setShowModalDelete(true);
+        }}
         modifiers={{
-          projectId: (x) => x.name,
+          projectId: (x) => x?.name,
           approved: (x) => (x ? 'Approved' : 'Not approved'),
           Task: (x) =>
```
Comment (iaracriscenti) [inline src/Components/Employee/Timesheet/index.jsx:105]: I would add these lines to sum only the worked hours that correspond to the timesheet period (in this case, 2022-07). Maybe later we can add an option to filter by month, and change currentDate value
```
const currentDate = new Date().toISOString().slice(0, 7);
```
```

Task: (x) =>
        x
          ?.filter((task) => task.taskId.createdAt.slice(0, 7) === currentDate)
          .reduce((previous, current) => {
            return previous + current.taskId.workedHours;
          }, 0),
```

**Review by iaracriscenti — COMMENTED** [review]
Add and delete Timesheet is working fine. I suggested a little change.

**Review by agusting19 — APPROVED** [review]
Looks nice and its working fine

**Review by ArianaMaldonado — APPROVED** [review]
**Test Result: Passed ✅**

**Pull request merged**